### PR TITLE
PSR-743 Don't POST heldProperty.otherDescription when empty

### PR DESCRIPTION
**Pull request by B-Savage**

### CODE REVIEW

File: app/transformations/LandOrPropertyTransactionsTransformer.scala [hunk 1]
```diff
@@ -548,7 +548,7 @@ class LandOrPropertyTransactionsTransformer @Inject() extends Transformer {
                       identityType = receivedLandType,
                       idNumber = None,
                       reasonNoIdNumber = None,
-                      otherDescription = Some(other.description)
+                      otherDescription = if (other.description.nonEmpty) Some(other.description) else None
```
Comment (cyeniceri) [inline app/transformations/LandOrPropertyTransactionsTransformer.scala:551]: Could be rewritten as 
`
Option.when(other.description.nonEmpty)(other.description)
`

**Review by cyeniceri — APPROVED** [review]

**Review by kberec — APPROVED** [review]

**Review by kberec — CHANGES_REQUESTED** [review]
Looks @harishmuddegowda found some issue when testing, after the data is posted and subsequently fetched, please refer to the slack conversation.

File: app/transformations/LandOrPropertyTransformer.scala [hunk 2]
```diff
@@ -211,7 +211,7 @@ class LandOrPropertyTransformer @Inject() extends Transformer {
               prop => {
                 OtherRecipientDetailsPage(srn, index, LandOrPropertySeller) -> RecipientDetails(
                   heldPropertyTransaction.optPropertyAcquiredFromName.get,
-                  prop.otherDescription.get
+                  prop.otherDescription.getOrElse(None)
```
Comment (cyeniceri) [inline app/transformations/LandOrPropertyTransformer.scala:214]: would it be better if we use :
prop.otherDescription instead prop.otherDescription.getOrElse(None)

Comment (cyeniceri) [reply app/transformations/LandOrPropertyTransformer.scala]: now it's changed :)

**Review by cyeniceri — APPROVED** [review]

**Review by kberec — APPROVED** [review]

**Pull request merged**